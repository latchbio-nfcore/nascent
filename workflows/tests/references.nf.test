def igenomes_base = 's3://ngi-igenomes/igenomes'

nextflow_pipeline {

    name "Test passing references"
    script "../../main.nf"

    // https://github.com/nf-core/nascent/issues/106
    test("Should work with BWA Index") {
        // FIXME PINTS Fails because it doesn't find anything.
        config './skip_pints.config'
        when {
            params {
                outdir = "$outputDir"
                aligner = "bwa"
                // TODO Update these to human
                bwa_index = "${igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/BWAIndex/version0.6.0/"
                fasta = "${igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Sequence/WholeGenomeFasta/genome.fa"
                gtf = "${igenomes_base}/Saccharomyces_cerevisiae/Ensembl/R64-1-1/Annotation/Genes/genes.gtf"
            }
        }

        then {
            assert workflow.success
            assert snapshot(
                workflow.trace.tasks().size(),
            ).match()
        }
    }

    // https://github.com/nf-core/nascent/issues/119
    test("Should work with gzipped references") {
        config './skip_gzip.config'
        when {
            params {
                outdir = "$outputDir"
                aligner = "bwa"
                fasta = 'https://raw.githubusercontent.com/nf-core/test-datasets/rnaseq/reference/genome.fasta.gz'
                gtf = 'https://raw.githubusercontent.com/nf-core/test-datasets/rnaseq/reference/genes.gtf.gz'
            }
        }

        then {
            assert workflow.success
            assert snapshot(
                workflow.trace.tasks().size(),
            ).match()
        }

    }

}
