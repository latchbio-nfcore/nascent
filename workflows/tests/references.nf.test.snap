{
    "Should work with gzipped references": {
        "content": [
            98
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-05T16:02:57.855321372"
    },
    "Should work with BWA Index": {
        "content": [
            113
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-05T16:01:56.042895683"
    }
}